npm_and_yarn/lib/dependabot/npm_and_yarn/registry_helper.rb

```diff
@@ -39,7 +39,13 @@ def find_corepack_env_variables
         registry_info = find_registry_and_token
 
         env_variables = {}
-        env_variables[COREPACK_NPM_REGISTRY_ENV] = registry_info[:registry] if registry_info[:registry]
+
+        if registry_info[:registry] # Prevent the https from being stripped in the process
+          registry = registry_info[:registry]
+          registry = "https://#{T.must(registry)}" unless T.must(registry).start_with?("http://", "https://")
```

Comment:
This assumes it'll always be `https` if not present... I'd argue it could just as likely be `http` if not present... 

Also, there may be other places in the code that this "bug" could bite us if we're stripping when we shouldn't be... 

Did you investigate where/why it's getting stripped in the first place?